deps/rabbit/src/rabbit_amqp_management.erl

```diff
@@ -147,29 +146,40 @@ handle_http_req(HttpMethod = <<"PUT">>,
             Result;
         {error, not_found} ->
             PermCache2 = check_dead_letter_exchange(QName, QArgs, User, PermCache1),
-            case rabbit_amqqueue:declare(
-                   QName, Durable, AutoDelete, QArgs, Owner, Username) of
-                {new, Q} ->
+            try rabbit_amqqueue:declare(
+              QName, Durable, AutoDelete, QArgs, Owner, Username) of
+              ARGS ->
```

Comment:
Why match with a variable here, and then check the variable with a case statement?

Reply:
Thank you @SimonUnge 
Solved here https://github.com/rabbitmq/rabbitmq-server/pull/11742/commits/f9707530b02a9018f0cbb1484b6c2b49a807e348